src/averageAges.js

```diff
@@ -59,6 +90,33 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const mothersNames = people
+    .filter(({ mother }) => mother)
+    .reduce((acc, person) => {
+      return acc.concat(person.mother);
+    }, []);
+
+  const children = people.filter(person =>
+    mothersNames.includes(person.mother));
+
+  const filteredChildren = onlyWithSon
+    ? children.filter(person => person.sex === 'm'
+      && mothersNames.includes(person.mother))
+    : children;
+
+  const ageDiffs = filteredChildren.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    return mother ? child.born - mother.born : 0;
+  });
+
+  const nonZeroAgeDiffs = ageDiffs.filter(diff => diff !== 0);
+
+  const totalAgeDiff = nonZeroAgeDiffs.reduce((acc, diff) => acc + diff, 0);
+
+  const averageAgeDiff = totalAgeDiff / nonZeroAgeDiffs.length;
```

Comment:
To simplify this you can filter your `filteredChildren` array and store there only children that have mother in array of people